bears/r/FormatRBear.py

```diff
@@ -68,24 +68,15 @@ def create_arguments(filename, file, config_file,
                 1
             ```
         """
-        rcode = ('library(formatR);'
-                 'formatR::tidy_source('
-                 'source="' + escape(filename, '"\\') + '",'
-                 'comment={r_keep_comments},'
-                 'blank={r_keep_blank_lines},'
-                 'arrow={r_use_arrows},'
-                 'brace.newline={r_braces_on_next_line},'
-                 'indent='
-                 '{tab_width}'.format(tab_width=tab_width,
-                                      r_keep_comments=_map_to_r_bool(
-                                          r_keep_comments),
-                                      r_keep_blank_lines=_map_to_r_bool(
-                                          r_keep_blank_lines),
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        options |= ({'width.cutoff=' + str(r_max_expression_length)}
+                    if r_max_expression_length != 0 else set())
+
+        rcode = 'library(formatR);formatR::tidy_source({})'.format(
+            ','.join(option for option in options))
```

Comment:
ah okay :)

2016-06-22 20:10 GMT+01:00 Mischa Krüger <notifications@github.com>:

> In bears/r/FormatRBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/525#discussion_r68116087>
> :
>
> > -                                      r_braces_on_next_line=_map_to_r_bool(
> > -                                          r_braces_on_next_line)))
> > -        # Disable r_max_expression_length if it is equal to 0
> > -        rcode += ('' if r_max_expression_length == 0
> > -                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
> > +        options = {'source="' + escape(filename, '"\\') + '"',
> > +                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
> > +                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
> > +                   'comment=' + _map_to_r_bool(r_keep_comments),
> > +                   'arrow=' + _map_to_r_bool(r_use_arrows),
> > +                   'indent=' + str(tab_width)}
> > +        options |= ({'width.cutoff=' + str(r_max_expression_length)}
> > +                    if r_max_expression_length != 0 else set())
> > +
> > +        rcode = 'library(formatR);formatR::tidy_source({})'.format(
> > +            ','.join(option for option in options))
>
> -->
>
> ','.join(options)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/525/files/3dbc2ae9d11ff726d82950116bc1543e121752b9#r68116087>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xohoXZKqSTcJd0-PfHXqOfZolzuCcks5qOYipgaJpZM4I14vI>
> .
>